Race condition in ini writing with concurrent use

When using `pydna` in multiple processes (eg: in tests with `pytest-xdist`), the ini file may attempt to be written multiple times or read before the default config is written.

Eg, this sequence will cause errors:
- process 1 opens the file in write mode `with open(_ini_path, 'w', encoding="utf-8") as f`
- process 2 reads the file `_parser.read(_ini_path)`
- process 1 writes the config `_parser.write(f)`

resulting in this error:
```
    from pydna.dseqrecord import Dseqrecord
../venv/lib/python3.10/site-packages/pydna/__init__.py:169: in <module>
    _mainsection = _parser["main"]
../python/lib/python3.10/configparser.py:964: in __getitem__
    raise KeyError(key)
E   KeyError: 'main'
```
(in process 2, which read the file before the contents were written)

---

As a fix, maybe it could use a `tempfile.NamedTemporaryFile` and then move it over at the end (which should be atomic on most platforms if we ensure the tempfile is in ini dir / same filesystem) or just skip creating the configuration file completely (eg: stop after `_parser["main"] = default_ini`)?

I'd be happy to PR either.

--- a/src/pydna/__init__.py
+++ b/src/pydna/__init__.py
@@ -139,6 +139,7 @@ import logging as _logging
 import logging.handlers as _handlers
 import appdirs as _appdirs
 import configparser as _configparser
+import tempfile as _tempfile
 from pydna._pretty import PrettyTable as _PrettyTable
 
 
@@ -183,8 +184,15 @@ if _ini_path.exists():
     _parser.read(_ini_path)
 else:  # otherwise it is created with default settings
     _parser["main"] = default_ini
-    with open(_ini_path, "w", encoding="utf-8") as f:  # TODO needs encoding?
-        _parser.write(f)
+    _temp_ini_file = _tempfile.NamedTemporaryFile(dir=_ini_path.parent, delete=False)
+    _temp_ini_path = _Path(_temp_ini_file.name)
+    try:
+        _temp_ini_file.close()
+        with _temp_ini_path.open("w", encoding="utf-8") as f:  # TODO needs encoding?
+            _parser.write(f)
+        _temp_ini_path.replace(_ini_path)
+    finally:
+        _temp_ini_path.unlink(missing_ok=True)
 
 # pydna related environmental variables are set
 # from pydna.ini if they are not set already

